Finna api

wontheone1 opened the pull request:
Use Finna API to fetch images

wontheone1 committed:
Use showing ranking instead of showing game

wontheone1 committed:
Change div id to app

wontheone1 committed:
Remove unnecessary comment

wontheone1 committed:
Turn off audio by default (annoying to hear it all the time)

wontheone1 committed:
Change button text to 'Play game'

wontheone1 committed:
Use reframe

wontheone1 committed:
Remove unused part

wontheone1 committed:
Separate view part

wontheone1 committed:
Render ranking-dashboard under root app component

wontheone1 committed:
Render reactively to the state of screen state

wontheone1 committed:
Show intro in the beginning

wontheone1 committed:
Add puzzle selection screen

wontheone1 committed:
Fetch finna img and display it

wontheone1 committed:
Pull various images from finna

wontheone1 committed:
Rename sync-puzzle-board to show side effect

wontheone1 committed:
Use .. macro

wontheone1 committed:
Bind phaser-loader

wontheone1 committed:
Make image source parameter of start-game!

And game starts when player chooses which image to play with

wontheone1 committed:
Sync comment to code

wontheone1 committed:
Do not store piece-x/y-scale get from phaser cache

wontheone1 committed:
Set util/puzzle-image-width/height with image of image-src

image-src is parameterized in start-game! function so we can set different sizes
depending on image-src

wontheone1 committed:
Refactor on-click of image list

wontheone1 committed:
Rename finna-img -> game-img

wontheone1 committed:
Set game-img when selecting image and

start the game when rendering game screen

wontheone1 committed:
Render game when game-img is set (as string for img url)

Note that Phaser renderer should be changed to Canvas
to allow using images from cross origin

wontheone1 committed:
Rename image-src-of -> set-game-image!

flyjwayur commented on build.boot:
```diff
@@ -23,6 +23,7 @@
                   [org.clojure/clojurescript "1.9.946"]
                   [org.clojure/core.async "0.4.474"]
                   [reagent "0.8.0-alpha2"]
+                  [re-frame "0.10.5"]
```
What is a term 'reaction' ? Is it from Reagent or Re-frame?:)

wontheone1 committed:
Rename screen->image-url

flyjwayur commented:
Excellent 👍

flyjwayur commented:
`(defn display-puzzle-selection-only-for-first-player []
  (if (not= nil (:sprites-state @util/game-state))
    util/show-game!
    util/show-puzzle-selection!))`

Could this code return proper fn as on-click fn to check whether it only displays the puzzle-selection page to the first player and the other players directly see the running game than choosing puzzle selection page? :)

wontheone1 committed:
Fix bug ranking dashboard was not showing

wontheone1 committed:
Fix bug when ranking board was shown before playing

The puzzle board is shown far below when ranking board was shown before.

Puzzle board is now created as soon as game starts before ranking button can be clicked

wontheone1 committed:
Extract set-on-click-callback-for-sprite! function

wontheone1 committed:
Extract puzzle-selection-view

wontheone1 committed:
Do not pass unnecessary functions to show-puzzle-board fn

wontheone1 committed:
Rename functions to show side-effect

wontheone1 committed:
Do not show control-buttons on game start
